test/component/login_test.dart

```diff
@@ -1,83 +1,163 @@
-    await tester.pumpAndSettle();
-
-    // Check for the logo on the Login Page
-    final logoFinder = find.byKey(LoginPage.logoKey);
-    expect(logoFinder, findsOneWidget);
-
-    // Check for the slogan on the Login Page
-    final sloganFinder = find.text(LoginPage.tagLineText);
-    expect(sloganFinder, findsOneWidget);
-
-    final loginButton = find.byKey(LoginButton.loginButtonKey);
-    // Check that the login button is displayed and contains the "Login" text
-    expect(
-      find.descendant(
-        of: loginButton,
-        matching: find.text("Sign in with Google"),
-      ),
-      findsOneWidget,
-    );
-
-    await tester.tap(loginButton);
-    await tester.pumpAndSettle();
-
-    // Check that pressing login redirects to the create account page
-    final createAccountPage = find.byType(CreateAccountPage);
-    expect(createAccountPage, findsOneWidget);
-
-    // And that pushing the confirm button redirects to the home page
-    final confirmButton = find.byKey(CreateAccountPage.confirmButtonKey);
-    await tester.tap(confirmButton);
-    await tester.pumpAndSettle();
-
-    // We must now be on the home page
-    final homePage = find.byType(HomePage);
-    expect(homePage, findsOneWidget);
+  group("Existing user data in repository testing", () {
+    final expectedUser = testingUserFirestore;
+
+    setUp(() async {
+      fakeFireStore = FakeFirebaseFirestore();
+      userCollection = fakeFireStore.collection(UserFirestore.collectionName);
+      userRepo = UserRepositoryService(
+        firestore: fakeFireStore,
+      );
+      await userCollection
+          .doc(expectedUser.uid.value)
+          .set(expectedUser.data.toDbData());
+    });
+
+    ProviderScope getMockedProxima() {
+      return ProviderScope(
+        overrides: [
+          ...firebaseAuthMocksOverrides,
+          userRepositoryProvider.overrideWithValue(userRepo),
+        ],
+        child: const ProximaApp(),
+      );
+    }
+
+    testWidgets("Login and get to home page flow", (tester) async {
+      await tester.pumpWidget(getMockedProxima());
+      await tester.pumpAndSettle();
+
+      final loginButton = find.byKey(LoginButton.loginButtonKey);
+      await tester.tap(loginButton);
+      //Needs a delay to allow the existance check to complete
+      await tester.pumpAndSettle(delayNeededForAsyncFunctionExecution);
+
+      final loginButtonStillHere = find.byType(HomePage);
+      expect(loginButtonStillHere, findsOneWidget);
+    });
   });
 
-  testWidgets("Login and Logout using create account page", (tester) async {
-    await tester.pumpWidget(mockedProxima);
-    await tester.pumpAndSettle();
-
-    final loginButton = find.byKey(LoginButton.loginButtonKey);
-    await tester.tap(loginButton);
-    await tester.pumpAndSettle();
-
-    final createAccountPage = find.byType(CreateAccountPage);
-    expect(createAccountPage, findsOneWidget);
-
-    final logoutButton = find.byKey(CreateAccountPage.logoutButtonKey);
```

Comment:
I don't see any equivalent test for the logout functionality. I think it was completely removed in this branch, and we should not remove tested functionality if it is still a feature of the application.

Can you please make this test the logout again ? Just add a logout button either on the overview or on the profile page.

The test name says "Login and Logout using create account page".

Reply:
Actually, we should have two tests for this now, one from the overview, and one for the intermediate page. Both should logout the user in some way.